runtime/reconcilers/project_parser.go

```diff
@@ -116,6 +117,23 @@ func (r *ProjectParserReconciler) Reconcile(ctx context.Context, n *runtimev1.Re
 		return runtime.ReconcileResult{Err: fmt.Errorf("failed to sync repo: %w", err)}
 	}
 
+	// Update the commit timestamp
+	ts, err := repo.CommitTimestamp(ctx)
+	if err != nil {
+		r.C.Logger.Error("failed to get commit timestamp", zap.String("error", err.Error()))
+	}
+	// Convert to timestamppb.Timestamp
+	timeStamp := timestamppb.New(ts)
+
+	// Update the commit timestamp
+	if pp.State.CurrentCommitOn != timeStamp {
+		pp.State.CurrentCommitOn = timeStamp
+		err = r.C.UpdateState(ctx, n, self)
```

Comment:
IMO it's fine to skip the check for whether the commit time changed, and just update both fields if the hash changed. The commit hash and timestamp will almost always change in unison.

Also, merging merging this update logic with the `UpdateState` call for `CurrentCommitSha` below would be nice as there would be one less DB write roundtrip.

Reply:
Sounds good to me, I will make that change